### ISSUE TITLE
TypeError: Cannot set property 'instanceId' of undefined

### ISSUE THREAD

**Issue by zmikolaj** [SocketCluster/sc-broker#2]
Hi, im use this coce:

``` javascript
var scBroker = require('sc-broker');
var dataServer = scBroker.createServer({port: 8008});

dataServer.on('ready', function () {
    console.log('   >> Data server ready');
    var client = scBroker.createClient({port: 8008});

    client.subscribe('testChannel', function (err) { console.log(err); });
    client.on('message', function(ch, data) {
        console.log(ch, data);
    });
    client.publish('testChannel', 'ELO', function(err) { console.log(err); });
});
```

I dont know what i doing wrong.

**Comment by wactbprot** [comment]
you do nothing wrong; this code works...

**Comment by icyhate** [comment]
this code does not work. what do you mean it works, am i missing something even the simple code
var scBroker = require('sc-broker');
module.exports.run = function (broker) {
    var dataServer = scBroker.createServer({port: 9000})
  console.log('   >> Broker PID:', process.pid);
};

gives me [Error] TypeError: Cannot set property 'instanceId' of undefined

**Comment by icyhate** [comment]
this library is definitly deprecated. 
i changed the options as so 

> >  var serverOptions = {
> >     id: options.id,
> >     brokerOptions: options.brokerOptions, // added this line
> >     debug: options.debug,
> >     socketPath: options.socketPath,
> >     expiryAccuracy: options.expiryAccuracy,
> >     downgradeToUser: options.downgradeToUser,
> >     brokerControllerPath: options.brokerControllerPath,
> >     initControllerPath: options.initControllerPath,
> >     processTermTimeout: options.processTermTimeout
> >   };`

and now i feed it the broker 
`var dataServer = scBroker.createServer({port: 8008, brokerOptions: broker});`

but i get an error still [Error] TypeError: Converting circular structure to JSON

i guess i will settle for using mongodb for my broker, its super shitty though hope  wactbprod helps

how did you get that code to run??

**Comment by wactbprot** [comment]
seems that a newer version introduces the problem; need some time to figure out. as a work around you can use version 2.1.0 (which is the version I use so far)

**Comment by wactbprot** [comment]
commit 38a8a648b56d0c7973fb5b65df126fd451e7d26f is the latest working; help is on the way

**Comment by jondubois** [comment]
@icyhate Can you paste the full error message showing the line numbers?

**Comment by wactbprot** [comment]
```
/.../brocker-tst/node_modules/sc-broker/server.js:125
  this.instanceId = this.options.instanceId;
                                ^

TypeError: Cannot read property 'instanceId' of undefined
    at EventEmitter.Broker (/...brocker-tst/node_modules/sc-broker/server.js:125:33)
    at initBrokerServer (/...brocker-tst/node_modules/sc-broker/server.js:163:14)
    at process.<anonymous> (/.../brocker-tst/node_modules/sc-broker/server.js:498:9)
    at emitTwo (events.js:106:13)
    at process.emit (events.js:191:7)
    at process.nextTick (internal/child_process.js:719:12)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
```

**Comment by jondubois** [comment]
Thanks @wactbprot :) I'll look into this tomorrow.

**Comment by jondubois** [comment]
@icyhate @wactbprot Ok, this has now been fixed in v2.1.4. Thanks.

**Comment by wactbprot** [comment]
test for 2.1.4 are working:

```
40 passing (2s)
```

coverage  is:

```
Statements   : 69.28% ( 309/446 )
Branches     : 46.91% ( 91/194 )
Functions    : 67.21% ( 41/61 )
Lines        : 69.28% ( 309/446 )
```

good job @jondubois :+1:
